modules/log/log.go

```diff
@@ -264,11 +264,13 @@ func (l *Logger) StartLogger() {
 	for {
 		select {
 		case bm := <-l.msg:
+			l.lock.Lock()
 			for _, l := range l.outputs {
 				if err := l.WriteMsg(bm.msg, bm.skip, bm.level); err != nil {
 					fmt.Println("ERROR, unable to WriteMsg:", err)
 				}
 			}
+			l.lock.Unlock()
```

Comment:
This one is a bit of tricky and I am afraid it would cause dead-lock if the select statement is included.